### ISSUE TITLE
config.yaml / customization / splash not working as expected (expectations and default values for configuration unclear)

### ISSUE THREAD

**Issue by Petrox** [inventree/InvenTree#5432]
### Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find a similar issue

### Describe the bug*

TLDR: no need to read it, solution has been found and PR incoming to add the RELATIVE TO comment to config template.
 

### Steps to Reproduce

docker based install
current stable image
config.yaml edited

 - good: customize / login_message works as expected
 - good: customize / navbar_message works as expected
 - ? splash: splash_screen.jpg   does not work as expected 

 1. default login screen has a background image in static/img/inventree_splash.jpg
 2. default config.yaml has a commented out    splash: splash_screen.jpg     (but there is no splash_screen.jpg in the repo AND the splash screen we see has a different name)
 3. I would like to override this with my image
 4. I copied my image to the static/img folder under the name "splash_screen.jpg"
 5. updated the config.yaml and uncommented the splash line. 
![image](https://github.com/inventree/InvenTree/assets/3036509/9bf255e2-6fa8-4ce1-b39f-a0b02eee905d)
6. restarted all containers (three times based on the expert advice found at https://youtu.be/uRGljemfwUE?t=59 )
7. Absolutely no change in the login screen aesthetics.

Conclusion: I must have missed something.

Google-fu: inventree splash, inventree configuration splash... found comment

![Image](https://github.com/inventree/InvenTree/assets/3036509/82f1eef0-34e7-4b76-a472-11b71248b5b1)

restart3x, splash screen appears.

Let's rewrite the whole github issue. 
Or just make it into a lesson learned and update the config template with this small note?

### Expected behaviour

uncommenting the splash field should look for the image in the same folder where the factory default splash image is stored.

### Deployment Method

- [X] Docker
- [ ] Bare metal

### Version Information

0.12.3

### Please verify if you can reproduce this bug on the demo site.

- [X] I can reproduce this bug on the demo site.

**Comment by SchrodingersGat** [comment]
@petrox the default behaviour (expectation) is that the system looks for custom images as specified relative to the `/static/` directory. You can put them in the `/img/` subdir if you like, but this is not required. I do no think that the suggest change is necessary.

If you put the image in the top-level `/static/` directory, and a matching value in your configuration file, I assume this works perfectly fine?

**Comment by Petrox** [comment]
I meant expectations and assumptions are mismatched.

So the program expects the image path to be relative to static. (Which is
perfectly fine for me and I don't think the code needs changing).

But the USER expects the commented out config directive to behave as it is
behaving by default and tries to change that.

No documentation or the config.yaml itself mentions what the default value
of that setting is and actually the examle in the config points to a
nonexistent file. (This can be improved).

Also as best practice the config file could "document itself", so the
relative to /static/ comment improves the configuration file.

My change does not change the behavior of the system or the default value
or the paths, it just shows the reader what is the default value for that
specific item and then the user may decide how to change it and it will
work on the first try.

My comments are not criticism, I'm just walking the path of the new user
and along the way when I see potential for improvement I will make the path
clearer for those coming after me.

Point is, since you have all the details and context in your mind, these
(very minor) things look trivial or invisible, while a newcomer could bump
his head into them.

The default splash is in /static/img so the new user should probably put
their splash there or anywhere else under static, the code does not need
any change.

**Comment by SchrodingersGat** [comment]
> No documentation or the config.yaml itself mentions what the default value of that setting is and actually the example in the config points to a nonexistent file. (This can be improved).

Ok, I see what you are getting at here. 

The issue is that placing a file in the `./img/` directory may be overwritten when the inventree installation is updated. I would suggest that we change the default value to something like `custom/splash.jpg` - indicating that the user should place in the custom subdirectory. This way, we can avoid any possibility of the system overwriting this file on an update

**Comment by matmair** [comment]
@SchrodingersGat should we use maintainers permissions to change the path and merge or let this close?
